platform/linux_platform.go

```diff
@@ -1469,16 +1408,20 @@ func (p linux) partitionDisk(availableSize uint64, desiredSwapSizeInBytes *uint6
 	var swapPartitionPath string
 	var dataPartitionPath string
 
+	if len(labelPrefix) >= 36 {
```

Comment:
Yes, GPT partition names can up to 36 characters. And 24 chars agent ID ends with a dash`-`. It can easy to append another part. 

I will refactor partition name format contains `bosh-partition` + <truncated-agent-uuid> + <paritioni-index> and concatenate with dash.